tests/test_packages/test_connections/test_p2p_libp2p/test_local_dht.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This test module contains integration tests for P2PLibp2p connection."""
+
+import itertools
+import json
+import os
+
+import pytest
+
+from aea.test_tools.test_cases import AEATestCaseMany
+
+from packages.valory.connections import p2p_libp2p, p2p_libp2p_client
+from packages.valory.connections.p2p_libp2p.connection import (
+    PUBLIC_ID as P2P_CONNECTION_PUBLIC_ID,
+)
+from packages.valory.connections.p2p_libp2p_client.connection import (
+    PUBLIC_ID as P2P_CLIENT_CONNECTION_PUBLIC_ID,
+)
+
+from tests.conftest import DEFAULT_LEDGER, UseACNWithBootstrappedEntryNodes
+from tests.test_packages.test_connections.test_p2p_libp2p.base import (
+    BaseP2PLibp2pTest,
+    LIBP2P_LEDGER,
+    libp2p_log_on_failure_all,
+    make_cert_request,
+    ports,
+)
+
+
+LOCAL_DHT_MADDRS = [
+    "/dns4/0.0.0.0/tcp/10000/p2p/16Uiu2HAm2yxmLQTZTrxjo5c4k5ka8AVMcpeD5zMMeasE6xDw1YQw",
+    "/dns4/0.0.0.0/tcp/10001/p2p/16Uiu2HAkw99FW2GKb2qs24eLgfXSSUjke1teDaV9km63Fv3UGdnF",
+    "/dns4/0.0.0.0/tcp/10002/p2p/16Uiu2HAm4aHr1iKR323tca8Zu8hKStEEVwGkE2gtCJw49S3gbuVj",
+][:1]
+LOCAL_DHT_DELEGATE_URIS = ["localhost:11000", "localhost:11001", "localhost:11002"][:1]
+LOCAL_DHT_PUBLIC_KEYS = [
+    "0270475f9b78c0285a6ac6067582f5e159ec147ccb03aee16a32731f68920b1ae8",
+    "02197b55d736bd242311aaabb485f9db40881349873bb13e8b60c8a130ecb341d8",
+    "0287ee61e8f939aeaa69bd7156463d698f8e74a3e1d5dd20cce997970f13ad4f12",
+][1:]
```

Comment:
ok this is incorrect. I was using this for testing purposes, slicing first item (bootstrap / genesis node) which was working, slicing last two (entry nodes) is how I want to be able to run the tests - but right now slicing is inconsistent....